server/src/main/java/org/apache/iotdb/db/qp/sql/IoTDBSqlVisitor.java

```diff
@@ -785,6 +787,126 @@ public Operator visitUnsetSchemaTemplate(IoTDBSqlParser.UnsetSchemaTemplateConte
     return operator;
   }
 
+  // Set Migration
+  @Override
+  public Operator visitSetMigration(IoTDBSqlParser.SetMigrationContext ctx) {
+    SetMigrationOperator operator = new SetMigrationOperator(SQLConstant.TOK_SET);
+
+    if (ctx.storageGroup != null) {
+      operator.setStorageGroup(parsePrefixPath(ctx.storageGroup));
+    }
+    if (ctx.ttl != null) {
+      operator.setTTL(Long.parseLong(ctx.ttl.getText()));
+    }
+    if (ctx.startTime != null) {
+      operator.setStartTime(parseDateFormat(ctx.startTime.getText()));
+    }
+    if (ctx.targetDir != null) {
+      FSFactory fsFactory = FSFactoryProducer.getFSFactory();
+      File targetDir = fsFactory.getFile(parseStringLiteral(ctx.targetDir.getText()));
+      if (!targetDir.exists()) {
+        throw new SQLParserException("unknown directory");
+      } else if (!targetDir.isDirectory()) {
+        throw new SQLParserException("not a directory");
+      }
+      operator.setTargetDir(targetDir);
+    }
```

Comment:
1. Storage group, ttl, start time and target dir cannot be null.
2. Add test to test null value.

Reply:
Added.